examples/test-exec-doc.yml

```diff
@@ -0,0 +1,221 @@
+# Name of the workflow
+name: Run Tests on Push to Main
+
+# Set permissions required for the workflow
+permissions:
+  contents: write           # Allows writing repository contents
+  pull-requests: write      # Allows creating and modifying pull requests
+  id-token: write           # Allows generating tokens for authentication
+  issues: write             # Allows creating and modifying issues
+
+# Define the events that trigger the workflow
+on:
+  push:
+    paths:
+      - '**/*.md'           # Trigger on push to any Markdown file
```

Comment:
So correct me if I'm wrong, but won't this action run on any push to any branch in their repository? I.E. If I push markdown to `vmarcella/testing` that has markdown file changes, won't this action run and create the pull request for me (Regardless of if the branch is ready for a pull request or not)?

Reply:
i didn't understand - this action is triggered from the branch where markdown changes are made. In your example, it would test the changed files compared to the last commit in vmarcella/testing and will only run in vmarcella/testing, not in all branches. If a PR already exists, it will update the PR. If it doesn't exist, it creates a new one

Reply:
That's not what I meant, what I was trying to get at is that any branch in the repo that is making changes which include markdown files will automatically have this action run and create a pull request for them, even if they're not done with working on that branch. If that is desired behavior for your example, then that is fine, but this will create PRs whenever someone adds any markdown content to their branch regardless of its executable or if they're done working on it.

Reply:
oh good point! let me make it so that the PR creation is an optional step that the customers can toggle on and off in the action itself